app/vminsert/influx/request_handler.go

```diff
@@ -105,9 +106,15 @@ func insertRows(at *auth.Token, db string, rows []parser.Row, mayOverrideAccount
 			ic.Labels = ic.Labels[:len(ic.Labels)-1]
 			ic.AddLabel("", metricGroup)
 			ic.MetricNameBuf = storage.MarshalMetricLabelRaw(ic.MetricNameBuf[:metricNameBufLen], placeholderLabel)
-			storageNodeIdx := ic.GetStorageNodeIdx(&atCopy, ic.Labels)
-			if err := ic.WriteDataPointExt(&atCopy, storageNodeIdx, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
-				return err
+			storageNodeGroupIDs := ic.GetStorageNodeGroupIds(&atCopy, ic.Labels)
+			for _, storageNodeGroupID := range storageNodeGroupIDs {
+				if err := ic.WriteDataPointExt(at, storageNodeGroupID, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
+					if ic.AllReplicationGroupsFailed(storageNodeGroupID.Group) {
```

Comment:
It is better counting errors returned from every call to `WriteDataPointExt` across all the replication groups and then logging the `All replication groups failed` error. Something like the following:

var errorsCount int
```go
for _, rg := range replicationGroups {
  ...
  if err := ic.WriteDataPointExt(....); err != nil {
    logger.Errorf("cannot write to replication group %q: %s", rg.name, err)
    errorsCount++
  }
}
if errorsCount == len(replicationGroups) {
    return fmt.Errorf("writing to all %d replication groups have failed", len(replicationGroups))
}
```